dodo.py

```diff
@@ -0,0 +1,325 @@
+        verbosity=0,  # stderr is too noisy by default
+    )
+
+
+def task_lark():
+    """Generate Lark grammar files based on keyword information in YAMLs."""
+    script = 'content/yaml_to_lark_utils.py'
+    keyword_yamls = glob('content/keywords/*.yaml')
+    grammars = ['grammars/keywords-' + replace_ext(path.basename(y), '.lark') for y in keyword_yamls]
+
+    return dict(
+        title=lambda _: 'Creating Lark grammar files',
+        file_dep=[
+            script,
+            'grammars/keywords-template.lark',
+            *keyword_yamls,
+        ],
+        actions=[
+            [python3, script],
+        ],
+        targets=grammars,
+    )
+
+
+def task_prefixes():
+    """Generate Python prefixes for TypeScript"""
+    script = 'build-tools/heroku/generate-prefixes-ts'
+
+    return dict(
+        title=lambda _: 'Generate Python prefixes for TypeScript',
+        file_dep=[
+            script,
+            *glob('prefixes/*.py'),
+        ],
+        actions=[
+            [script],
+        ],
+        targets=[
+            'static/js/pythonPrefixes.ts'
+        ],
+    )
+
+
+def task_lezer_parsers():
+    """Generate Lezer parsers."""
+    script = 'build-tools/heroku/generate-lezer-parsers'
+    grammars = glob('highlighting/lezer-grammars/level*.grammar')
+    lezer_files = []
+    for grammar in grammars:
+        base = replace_ext(path.basename(grammar), '')
+        lezer_files.append(f'static/js/lezer-parsers/{base}-parser.ts')
+        lezer_files.append(f'static/js/lezer-parsers/{base}-parser.terms.ts')
+
+    return dict(
+        title=lambda _: 'Generate Lezer parsers',
+        file_dep=[
+            *grammars,
+            script,
+        ],
+        actions=[
+            [script],
+        ],
+        targets=lezer_files,
+    )
+
+
+def task_extract():
+    """Extract new translateable keys from the code."""
+    return dict(
+        title=lambda _: 'Extract new keys from the code',
+        actions=[
+            'pybabel extract -F babel.cfg -o messages.pot . --no-location --sort-output',
+            'pybabel update -i messages.pot -d translations -N --no-wrap',
+        ],
+    )
+
+
+######################################################################################
+# Some useful task groups
+#
+
+def task_backend():
+    """Run all tasks necessary to prepare the backend."""
+    return dict(
+        actions=None,
+        task_dep=[
+            'compile_babel',
+            'generate_static_babel_content',
+            'lark',
+        ],
+    )
+
+
+def task_frontend():
+    """Run all tasks necessary to prepare the frontend."""
+    return dict(
+        actions=None,
+        task_dep=[
+            'tailwind',
+            'typescript',
+        ]
+    )
+
+
+def task_deploy():
+    """Commands to run at deploy time on Heroku.
+
+    This groups other commands.
+    """
+    return dict(
+        actions=None,
+        task_dep=['frontend', 'backend']
```

Comment:
so this means that the `deploy` task would run both of the `frontend` and `backend` tasks? That's actually very nice for organization purposes!!

Reply:
That's what I was thinking, yes. 

On deploy on a Heroku machine we would run the `deploy` task, and that would just do "everything" that's necessary to get both the backend and frontend in order.